packages/git/src/browser/git-repository-provider.ts

```diff
@@ -38,6 +42,22 @@ export class GitRepositoryProvider {
     }
 
     protected async initialize(): Promise<void> {
+        const workspaceDescriptionData = await this.workspaceService.workspaceDescription;
+        this.root = workspaceDescriptionData.root;
+        this.roots = workspaceDescriptionData.roots;
+        this.workspaceService.onWorkspaceChanged(async event => {
+            if (event) {
+                this.root = event.root;
+                this.roots = event.roots;
+                if (this.selectedRepository) {
```

Comment:
it is already covered in `refresh`, we just should call `refersh` in `onWorkspaceChanged`.